### PR TITLE
[Store onboarding] Hide other upsells from dashboard

### LINKED ISSUE #8971: Dashboard: hide other upsells (products or announcements) when the onboarding view is shown
Ref: pe5sF9-1ba-p2

> Q: Are we hiding any announcement or any other banners from this screen while the onboarding task list is being displayed?
Yes, if the onboarding task is being displayed, we will remove any other promotional banners in this screen.

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #8971 

## Description
Hide other announcements and banners when displaying store onboarding task list in dashboard.

## Testing instructions

**Prerequisite**
- Create a JN site without Jetpack. 

**Store with pending onboarding tasks**
- Launch the app and login into the above-created store
- Verify that you don't see any other announcements on the dashboard screen
- Verify that you can see the store onboarding task list on the dashboard screen

**Prerequisite**
Finish the pending onboarding tasks from `wp-admin` for the JN site
- Add a product
- Enable cash on delivery (WooCommerce -> Settings -> Payments)

**Store with no pending onboarding tasks**
- Launch the app and login into a store with no pending onboarding tasks
- Verify that you don't see any other announcements on the dashboard screen
- You shouldn't see "Add product"/

**WPCOM/Jetpack connected stores**
- [x] @selanthiraiyan to test the above scenario for WPCOM/Jetpack connected stores

## Screenshots

| Pending tasks | No pending tasks |
|--------|--------|
| Simulator Screen Shot - iPhone 14 - 2023-03-20 at 12 44 38 | Simulator Screen Shot - iPhone 14 - 2023-03-20 at 12 47 13 |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift [hunk 1]
```diff
@@ -487,8 +487,10 @@ private extension DashboardViewController {
     }
 
     func observeShowWebViewSheet() {
-        viewModel.$showWebViewSheet.sink { [weak self] viewModel in
-            guard let self = self else { return }
+        Publishers.CombineLatest(viewModel.$showWebViewSheet,
+                                 viewModel.$showOnboarding)
+        .sink { [weak self] viewModel, showOnboarding in
+            guard let self = self, !showOnboarding else { return }
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift]: ❓ is this change necessary? since this is to present a webview sheet, this doesn't conflict with the header view from my understanding 🤔

File: WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift [hunk 2]
```diff
@@ -618,8 +626,15 @@ private extension DashboardViewController {
     /// Shows or hides the free trial banner.
     ///
     func observeFreeTrialBannerVisibility() {
-        viewModel.$freeTrialBannerViewModel.sink { [weak self] viewModel in
+        Publishers.CombineLatest(viewModel.$freeTrialBannerViewModel,
+                                 viewModel.$showOnboarding)
+        .sink { [weak self] viewModel, showOnboarding in
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift]: ❓ similar to above, from my understanding we just want to hide other header views above the onboarding card. the bottom banner shouldn't be affected

File: WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift [hunk 3]
```diff
@@ -823,28 +838,35 @@ private extension DashboardViewController {
     }
 
     func observeBottomJetpackBenefitsBannerVisibilityUpdates() {
-        Publishers.CombineLatest(ServiceLocator.stores.site, $dashboardUI.eraseToAnyPublisher())
-            .sink { [weak self] site, dashboardUI in
-                guard let self = self else { return }
+        Publishers.CombineLatest3(ServiceLocator.stores.site,
+                                  $dashboardUI.eraseToAnyPublisher(),
+                                  viewModel.$showOnboarding)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift]: ❓ similar to above, I don't think the bottom banner is affected from my understanding.

**Review by jaclync — COMMENTED** [review]
I think I had a different understanding about the changes, lemme know if I missed something 🤔

**Comment by selanthiraiyan** [comment]
> I think I had a different understanding about the changes, lemme know if I missed something 🤔

I have updated the code to hide only the announcements. (Top portion of the screen)

| Before | After |
|--------|--------|
| Before | After |


I have also asked about this to the team. p1679388559162639-slack-C045CUK1Y3U
I will hide the other banners if I get confirmation. (In a different PR) 

Kindly take another look.

**Review by jaclync — APPROVED** [review]
LGTM :shipit: thanks for starting the discussion!
